### DISCUSSION TITLE
HMIP-WTH-2 nach Batteriewechsel nicht mehr steuerbar

### DISCUSSION THREAD

**Discussion by jaschkopf** [OpenCCU/OpenCCU#1960] [🇩🇪 Fragen&Antworten]
### Describe the issue you are experiencing

Nachdem ich bei einem meiner HMIP-WTH-2 die Batterie tauschen musste, kann das Gerät nicht mehr über die Zentrale gesteuert werden --> Servicemeldung Gerätekommunikation gestört. Wenn ich am Thermostat die Temperatur ändere, wird diese jedoch problemlos an die Zentrale übertragen. Die Übertragung von der Zentrale ans Thermostat wird dann aber trotzdem nicht durchgeführt. 

### Describe the behavior you expected

Das Gerät soll sich wieder normal über die Zentrale steuern lassen.

### Steps to reproduce the issue

1. Batterie entfernen
2. Batterie wieder einsetzen
3. Kommunikation von Zentrale zum Thermostat ist gestört
4. Thermostat kann trotzdem an die Zentrale senden
5. Antennensymbol im Display des Thermostat blinkt
...

### What is the version this bug report is based on?

3.63.9.20220521

### Which base platform are you running?

ova (Open Virtual Infrastructure)

### Which HomeMatic/homematicIP radio module are you using?

RPI-RF-MOD

### Anything in the logs that might be useful for us?

```txt
Für mich sehen die Logs erstmal unauffällig aus. Lediglich dieser Eintrag von meinem HMIP-FALMOT-C12 ist auffällig, aber vermutlich in keiner Verbindung zu meinem Problem:
May 29 13:07:25 de.eq3.cbcs.server.core.framehandling.HMIPApplicationHandler WARN  [vert.x-eventloop-thread-0] AP 3014F711A0001F58A9A70AE6: Device 3014F711A0001B9BE9A04B72 send BINARY_GROUP status for invalid channel with index 17 
May 29 13:07:25 de.eq3.cbcs.server.core.framehandling.HMIPApplicationHandler WARN  [vert.x-eventloop-thread-0] AP 3014F711A0001F58A9A70AE6: Device 3014F711A0001B9BE9A04B72 send BINARY_GROUP status for invalid channel with index 21
```

### Additional information

Ich habe bereits ein Downgrade auf verschiedene ältere Versionen (3.61.7.20220226, 3.61.7.20211218) probiert, jedoch ohne Erfolg. Das Thermostat wurde mehrfach auf Werkseinstellung zurück gesetzt, sowie aus der Zentrale gelöscht, ebenfalls ohne Erfolg. Die Zentrale bzw. der Server auf dem die Proxmox VM läuft wurden ebenfalls mehrfach neu gestartet. Auch ohne Erfolg.

Der Fehler lässt sich mit weiteren Thermostaten reproduzieren. Nachdem die Batterie entfernt wurde, keine Verbindung mehr von der Zentrale zum Thermostat. Alle Thermostate haben die Firmware 2.6.0

**Comment by Baxxy13** [comment]
Da ich heute mal bei einem WTH-2 die Batterien wechseln musste habe ich dabei genauer hingesehen.
Nachdem das WTH-2 gebootet hatte blinkte für etwa 5s das Antennensymbol und leuchtete dann dauerhaft. 
Anschließend ließ es sich problemlos aus der WebUI heraus bedienen. 
Ich kann das somit nicht reproduzieren und darum ist es eher unwahrscheinlich das es ein Bug in RaspberryMatic ist.

WTH-2 / FW: 2.6.0 / RM.ova 3.63.9.20220521 mit RPI-RF-MOD auf HB-RF-USB-2
Direktverknüpft mit dem WTH-2 sind ein HmIP-SWDM und eine HmIP-PS

**Comment by jaschkopf** [comment]
Ich habe heute nochmal frisch eine VM mit dem aktuellsten OVA Image aufgesetzt und die Thermostate lassen sich trotzdem nicht steuern. Scheinbar liegt es an den Thermostaten selbst. Um den Fehler zu verifizieren habe ich bei einem 3. Thermostat die Batterien kurz raus genommen. Vorher lies es sich einwandfrei aus dem WebUI steuern, nach dem wieder einsetzen kann es nicht mehr erreicht werden. Die Übertragung vom Thermostat an die Zentrale funktioniert aber nach wie vor. Wenn das so weiter geht, werden meine Thermostate nach und nach unbrauchbar. 

Ich werde mal kontakt zu EQ3 aufnehmen, mal sehen was die zu dem Problem sagen.

**Comment by jaschkopf** [comment]
Ich habe gerade nochmal eine Interessante entdeckung gemacht. Nach einem neustart der Zentrale und dem Versuch eines der betroffenen Thermostate zu steuern fand ich diese Errors im Log: 

>[CallXmlrpcMethod():iseXmlRpc.cpp:2613]
>Jun  1 17:34:58 homematic-raspi local0.debug ReGaHss: Verbose: call on change; id = 1538 [SetLastDPActionTime():iseDOMchn.h:288]
>Jun  1 17:34:58 homematic-raspi local0.debug ReGaHss: Verbose: call on change; id = 1514 [SetLastActionTime():iseDOMdev.h:225]
>Jun  1 17:34:58 homematic-raspi local0.info ReGaHss: Info: read flag is not set; operations = 2 [ReadValue():iseDOMdpHSS.cpp:133]
>Jun  1 17:34:58 homematic-raspi local0.info ReGaHss: Info: read flag is not set; operations = 6 [ReadValue():iseDOMdpHSS.cpp:133]
>Jun  1 17:34:58 homematic-raspi local0.info ReGaHss: Info: interface name= HmIP-RF for ID=1009 [GetInterfaceByID():iseXmlRpc.cpp:2552]
>Jun  1 17:34:58 homematic-raspi local0.warn ReGaHss: WARNING: XMLRPC 'getValue': rpcClient.isFault() failed (url: xmlrpc://127.0.0.1:32010, params: {"000A9A4996AC7D:1","PARTY_TIME_START"}, result: [faultCode:-5,faultString:"Unknown Parameter value for value key: PARTY_TIME_START"]) [CallXmlrpcMethod():iseXmlRpc.cpp:2608]
>Jun  1 17:34:58 homematic-raspi local0.err ReGaHss: ERROR: XMLRPC 'getValue' call failed (interface: 1009, params: {"000A9A4996AC7D:1","PARTY_TIME_START"}) [CallGetValue():iseXmlRpc.cpp:1435]
>Jun  1 17:34:58 homematic-raspi local0.err ReGaHss: ERROR: CallGetValue failed; sVal =  [ReadValue():iseDOMdpHSS.cpp:124]
>Jun  1 17:34:58 homematic-raspi local0.info ReGaHss: Info: interface name= HmIP-RF for ID=1009 [GetInterfaceByID():iseXmlRpc.cpp:2552]
>Jun  1 17:34:58 homematic-raspi local0.warn ReGaHss: WARNING: XMLRPC 'getValue': rpcClient.isFault() failed (url: xmlrpc://127.0.0.1:32010, params: {"000A9A4996AC7D:1","PARTY_TIME_END"}, result: [faultCode:-5,faultString:"Unknown Parameter value for value key: PARTY_TIME_END"]) [CallXmlrpcMethod():iseXmlRpc.cpp:2608]
>Jun  1 17:34:58 homematic-raspi local0.err ReGaHss: ERROR: XMLRPC 'getValue' call failed (interface: 1009, params: {"000A9A4996AC7D:1","PARTY_TIME_END"}) [CallGetValue():iseXmlRpc.cpp:1435]
>Jun  1 17:34:58 homematic-raspi local0.err ReGaHss: ERROR: CallGetValue failed; sVal =  [ReadValue():iseDOMdpHSS.cpp:124]
>Jun  1 17:34:58 homematic-raspi local0.info ReGaHss: Info: interface name= HmIP-RF for ID=1009 [GetInterfaceByID():iseXmlRpc.cpp:2552]
>Jun  1 17:34:58 homematic-raspi local0.warn ReGaHss: WARNING: XMLRPC 'getValue': rpcClient.isFault() failed (url: xmlrpc://127.0.0.1:32010, params: {"000A9A4996AC7D:1","PARTY_SET_POINT_TEMPERATURE"}, result: 
>[faultCode:-5,faultString:"Unknown Parameter value for value key: PARTY_SET_POINT_TEMPERATURE"]) [CallXmlrpcMethod():iseXmlRpc.cpp:2608]
>Jun  1 17:34:58 homematic-raspi local0.err ReGaHss: ERROR: XMLRPC 'getValue' call failed (interface: 1009, params: {"000A9A4996AC7D:1","PARTY_SET_POINT_TEMPERATURE"}) [CallGetValue():iseXmlRpc.cpp:1435]
>Jun  1 17:34:58 homematic-raspi local0.err ReGaHss: ERROR: CallGetValue failed; sVal = 0.000000 [ReadValue():iseDOMdpHSS.cpp:124]
>Jun  1 17:34:58 homematic-raspi local0.info ReGaHss: Info: interface name= HmIP-RF for ID=1009 [GetInterfaceByID():iseXmlRpc.cpp:2552]
>Jun  1 17:34:58 homematic-raspi local0.info ReGaHss: Info: XMLRPC 'getValue': rpcClient.execute() succeeded (url: xmlrpc://127.0.0.1:32010, params: {"000A9A4996AC7D:1","PARTY_MODE"}, result: false) 
>[CallXmlrpcMethod():iseXmlRpc.cpp:2613]
>Jun  1 17:34:58 homematic-raspi local0.debug ReGaHss: Verbose: call on change; id = 1538 [SetLastDPActionTime():iseDOMchn.h:288]
>Jun  1 17:34:58 homematic-raspi local0.debug ReGaHss: Verbose: call on change; id = 1514 [SetLastActionTime():iseDOMdev.h:225]
>Jun  1 17:34:58 homematic-raspi local0.info ReGaHss: Info: http id #1 sends parsed file [ParseFileAndSendWebResponse():httpServer.cpp:2091]
>Jun  1 17:35:00 homematic-raspi local0.info ReGaHss: Info: start web processing, worker thread #0 {"HTTP-Listener"} [spawnNewProcess():httpListener.cpp:208]
>Jun  1 17:35:00 homematic-raspi local0.info ReGaHss: Info: recvd 797 bytes by web server #1 [ThreadFunction():httpServer.cpp:795]
>Jun  1 17:35:00 homematic-raspi local0.info ReGaHss: Info: IseSession GetSessionId from URL: /esp/system.htm?sid=@bLn0Hu1hs9@&action=UpdateUI [GetSessionId():iseSession.cpp:185]
>Jun  1 17:35:00 homematic-raspi local0.info ReGaHss: Info: http id #1 sends parsed file [ParseFileAndSendWebResponse():httpServer.cpp:2091]
>Jun  1 17:35:00 homematic-raspi local0.info ReGaHss: Info: start web processing, worker thread #0 {"HTTP-Listener"} [spawnNewProcess():httpListener.cpp:208]
>Jun  1 17:35:00 homematic-raspi local0.info ReGaHss: Info: recvd 1535 bytes by web server #1 [ThreadFunction():httpServer.cpp:795]
>Jun  1 17:35:00 homematic-raspi local0.info ReGaHss: Info: IseSession GetSessionId from URL: /tclrega.exe [GetSessionId():iseSession.cpp:185]
>Jun  1 17:35:00 homematic-raspi local0.info ReGaHss: Info: interface name= HmIP-RF for ID=1009 [GetInterfaceByID():iseXmlRpc.cpp:2552]
>Jun  1 17:35:00 homematic-raspi local0.warn ReGaHss: WARNING: XMLRPC 'getValue': rpcClient.isFault() failed (url: xmlrpc://127.0.0.1:32010, params: {"001F58A9A70AE6:0","CARRIER_SENSE_LEVEL"}, result: [faultCode:-5,faultString:"Unknown Parameter value for value key: CARRIER_SENSE_LEVEL"]) [CallXmlrpcMethod():iseXmlRpc.cpp:2608]
>Jun  1 17:35:00 homematic-raspi local0.err ReGaHss: ERROR: XMLRPC 'getValue' call failed (interface: 1009, params: {"001F58A9A70AE6:0","CARRIER_SENSE_LEVEL"}) [CallGetValue():iseXmlRpc.cpp:1435]
>Jun  1 17:35:00 homematic-raspi local0.err ReGaHss: ERROR: CallGetValue failed; sVal = 0.000000 [ReadValue():iseDOMdpHSS.cpp:124]
>Jun  1 17:35:00 homematic-raspi local0.info ReGaHss: Info: http id #1 sends parsed file [ParseFileAndSendWebResponse():httpServer.cpp:2091]
>Jun  1 17:35:00 homematic-raspi local0.info ReGaHss: Info: start web processing, worker thread #0 {"HTTP-Listener"} [spawnNewProcess():httpListener.cpp:208]
>Jun  1 17:35:00 homematic-raspi local0.info ReGaHss: Info: recvd 653 bytes by web server #1 [ThreadFunction():httpServer.cpp:795]
> Jun  1 17:35:00 homematic-raspi local0.info ReGaHss: Info: IseSession GetSessionId from URL: /tclrega.exe [GetSessionId():iseSession.cpp:185]

**Comment by Baxxy13** [comment]
Was sagt denn eigentlich dein Carrier Sense Wert auf der Startseite?

Das ganze klingt irgendwie nach nem Funkstörer der verhindert das die Zentrale zu ihren Geräten sprechen kann.

**Comment by jaschkopf** [comment]
CarrierSense ist immer sehr niedrig, ich habe nie größere Werte als 5 dort gesehen. Das PRoblem besteht auch wenn ich mit dem Thermostat direkt neben der Zentrale stehe...

Gerade habe ich eine Uralte Version installiert, RaspberryMatic-3.49.17.20191225.ova auf einer frischen VM installiert, damit lässt sich das Thermostat wieder steuern. Vielleicht doch ein Bug?

**Comment by jaschkopf** [comment]
Ich habe jetzt die neue VM von 3.49.17.20191225 wieder auf 3.63.9.20220521 aktualisiert. Das Thermostat lies sich danach weiterhin normal steuern. Nachdem ich die Batterie raus und wieder rein gemacht habe, war der Fehler wieder da... 

EDIT: Auf Version 3.49.17.20191225 ist der Fehler nicht reproduzierbar gewesen. Batterie raus und wieder rein, Thermostat verbindet sich sofort und ist erreichbar.

**Comment by Baxxy13** [comment]
Sehr merkwürdig. 
Wie gesagt kann ich das bei mir nicht reproduzieren. 
Auch im Homematic-Forum hat bisher noch niemand von solch einem Problem berichtet.
Kann ja eigentlich nur was mit dem Funkmodul oder dessen Anbindung sein. 
Was nutzt du denn? HB-RF-USB oder HB-RF-ETH?

**Comment by jaschkopf** [comment]
Ich nutze die HB-RF-USB in Verbindung mit dem RPI-RF-MOD Modul. 

Definitiv ein seltsames Phänomen. Ich bin im Austausch mit eq3 und die haben es mittlerweile ebenfalls als Fehler der RaspberryMatic eingestuft.

Ich bin jetzt mittlerweile auf Version 3.61.5.20211113 gelandet, da ließen sich die Thermostate ebenfalls problemlos verbinden. Daher muss es ja etwas mit der Software zu tun haben. 

Gruß Jascha

**Comment by jp112sdl** [comment]
Hast du die 3.63.**8**.20220330 (https://github.com/jens-maus/RaspberryMatic/releases/tag/3.63.8.20220330) auch mal getestet?
Nur um mal zu prüfen, ob es mit den RPI-RF-MOD Firmware Updates was zu tun haben könnte.
Die 3.61.5 hatte diese glaub ich von 4.2.14 auf 4.4.12 geupdatet und die blieb bis zur 3.63.**9** wo dann jetzt die 4.4.22 kam

**Comment by jaschkopf** [comment]
Wird denn mit einem downgrade der RaspberryMatic auch automatisch die Firmware des RPI-RF-MOD geflashed? Oder wie läuft das ab? 

Ich kann die andere Version die Tage nochmal drauf spielen. Mit proxmox ist das ja schnell erledigt.

**Comment by jp112sdl** [comment]
> auch automatisch die Firmware des RPI-RF-MOD geflashed

Ja wird es

**Comment by jaschkopf** [comment]
Kurzes Update:  Ich habe eben das Upgrade auf 3.65.6.20220723 probiert, jedoch taucht der Fehler dann wieder auf. Bin momentan auf 3.65.5.20211113, damit läuft alles einwandfrei. 

Wäre es möglich, dass es ein Fehler in der Firmware der Thermostate selbst ist? Diese habe ich über RaspberryMatic auf 2.6.0 geflashed. Kann es sein, dass die Firmware beim Download beschädigt wurde und somit fehlerhaft auf all meinen Thermostaten installiert wurde? 

Kann man die Firmware der Thermostate downgraden oder manuell eine andere Version flashen?

**Comment by jaschkopf** [comment]
Nochmal ein Update: Ich habe gerade die 3.65.8.20220831 installiert. Der Fehler besteht weiterhin! Jedoch wurde mir ein FW Update der Thermostate von 2.6.0 auf 2.8.2 angeboten. Habe mich nicht getraut es zu installieren, nicht das meine Thermostate dadurch unbrauchbar werden... Bin jetzt wieder zurück auf 3.65.5.20211113. Hat jemand noch einen Vorschlag was ich probieren kann?

**Comment by bigtruite** [comment]
Habe auch dasselbe Problem. Habe mir das WTH-2 besorgt und heute über die Zentrale lokal eingebunden. Antennensymbol blinkt und es zeigt in der Zentrale immer wieder mal "Gerätekommunikation gestört" an. Dasselbe Problem wie in Komment 1. Löschen, und neu anlernen hat nichts gebracht. Bin auf 3.65.8.20220831 ova. Ein Update ist mir noch nicht angeboten worden. Bin noch auf 2.6.9.

**Comment by bigtruite** [comment]
So. Nach einem Neustart geht alles. Antennensymbol blinkt nicht mehr, sondern ist ständig vorhanden. Kommunikation in beiden Richtungen möglich... Komisch...

**Comment by jaschkopf** [comment]
Ich konnte das Problem gerade endlich beheben. Es scheint an der CCU bzw. dem RPI-RF-MOD gelegen zu haben. Ich habe neue Hardware gekauft (RPI-RF-MOD + HB-RF-USB-2) und eine neue VM mit der aktuellen 3.65.11.20221005.ova aufgesetzt. Im Anschluss habe ich mein Backup geladen und das Problem ist verschwunden. Somit muss es an der Hardware der CCU gelegen haben, da ein neu aufsetzen mit der alten Hardware (RPI-RF-MOD + HB-RF-USB) keinen Effekt hatte. Trotzdem allen die mitgewirkt haben, vielen Dank für die Hilfe! 👍

**Comment by Jens-La** [comment]
Hallo zusammen,

ich habe das gleiche Problem. Ich habe einen RPI3 B mit HM-MOD-RPI-PCB . Alle meine bisherigen Homematic IP Geräte (Thermostate + Rauchmelder) funktionieren ohne Probleme.

Jetzt versuche ich gerade ein "HmIP-WTH-2" zu betreiben. Das Anlernen hat noch geklappt, aber ab dann geht er alle paar Sekunden auf gestört.
Bei einem Funktionstest bleibt es häufig auf gelb, wird aber auch manchmal grün.
RPI läuft auf der Version: 3.65.11.20221218
HmIP-WTH-2 aktuell auf 2.6.0. Version 2.8.2 wird mir angeboten.

Ich hoffe Ihr könnt mir einen Tip geben.

Ich habe mir gerade mal die messages log datei angesehen. 
[homematic.log](https://github.com/jens-maus/RaspberryMatic/files/10324259/homematic.log)

Da scheint nicht wirklich viel zu gehen. Gefühlt geht es nach einem Homamativ neustart für die ersten Sekunden / Minuten.
